### ISSUE TITLE
Object.assign is missing in node 4/5/6 typings

### ISSUE THREAD

**Issue by louy** [types/env-node#22]

**Comment by felixfbecker** [comment]
`Object.assign` was added in 5. Should we include it in 6 though? Node 6 is 100% ES6 compatible, so users can use target: ES6 and use es6.lib

**Comment by louy** [comment]
Apparently it's supported in node 4.

```
louy@mb ~$ nvm use 4.0
Now using node v4.0.0 (npm v2.14.2)
louy@mb ~$ node -e 'console.log(Object.assign({}))'
{}
```

Also, these typings shouldn't rely on ts build target as the end goal is to support the `noLib` option. E.g. #14.

**Comment by felixfbecker** [comment]
Sorry, my bad. But doesn't it make errors when lib and Node typings define the same method with the same signature?

**Comment by louy** [comment]
It doesn't for functions. It does for properties, but there's a workaround (see #12).

**Comment by blakeembrey** [comment]
I don't think we should be supporting `noLib` (it's a lot to replicate and TypeScript's domain), but definitely at the minimum supporting `es{5,6}.core.d.ts`.

**Comment by felixfbecker** [comment]
@louy what is the workaround?
